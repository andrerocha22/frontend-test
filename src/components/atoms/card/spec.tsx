import React from 'react';
import { shallow, ShallowWrapper, mount } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import 'jest-styled-components';

import Card from './index';

const defaultProps = {
  primaryInfo: 'Test primary information',
  amount: '$999999',
  secondaryInfo: 'Test secondary information',
};

const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<Card {...setupProps} />);
};

const setupMount = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return mount(<Card {...setupProps} />);
};

describe('test card', () => {
  let component: ShallowWrapper = undefined;

  test('renders without error', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-card');
    expect(component.length).toBe(1);
  });

  test('card snapshot test ', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-card');
    expect(component).toMatchSnapshot();
  });

  test('card test props', () => {
    const wrapper = setupMount();
    expect(wrapper.prop('primaryInfo')).toEqual(defaultProps.primaryInfo);
    expect(wrapper.prop('amount')).toEqual(defaultProps.amount);
    expect(wrapper.prop('secondaryInfo')).toEqual(defaultProps.secondaryInfo);
  });
});
