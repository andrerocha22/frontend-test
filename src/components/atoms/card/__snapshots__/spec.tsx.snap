// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card card snapshot test  1`] = `
<styled.article
  data-test="component-card"
>
  <styled.div>
    <styled.p>
      Test primary information
    </styled.p>
    <styled.p>
      $999999
    </styled.p>
  </styled.div>
  <styled.p>
    Test secondary information
  </styled.p>
</styled.article>
`;
